python_modules/dagster/dagster/_core/storage/config.py

```diff
@@ -1,7 +1,27 @@
+# pyright: strict
```

Comment:
what is our criteria for turning on strict in general? Should we be pushing everyone to do this?

Reply:
See my longer response to review.